app/src/main/java/com/example/hris/ui/fragments/timelogs/AddTimeLogFragment.kt

```diff
@@ -43,17 +46,20 @@ class AddTimeLogFragment : Fragment() {
         }
 
         viewModel.loadingDialogState.observe(viewLifecycleOwner) {
-            (activity as MainActivity).setLoadingDialog(it)
+            mainViewModel.apiBool.value = it
         }
 
         viewModel.callValue.observe(viewLifecycleOwner) {
-            Toast.makeText(requireContext(), it.message, Toast.LENGTH_SHORT).show()
-            if (it.status == "0") {
-                val action =
-                    AddTimeLogFragmentDirections.actionAddTimeLogsFragmentToAddTimeLogSuccessFragment(
-                        binding.Spinner.selectedItem.toString()
-                    )
-                findNavController().navigate(action)
+            if (it != null) {
```

Comment:
Replace with `if (it ==null) return` so you won't have to enclose the rest of the block in braces thus reducing the indentation